### PR TITLE
chore(tasks): Test project fixture rebuild updates

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**WIP**

**Changes**
1. Uses git to track the steps of the fixture rebuild process
2. Keeps track of the last temporary project directory automatically
3. Introduces a basic TUITask which can handle task lists - similar to listr

**Todo**
1. Perform some more testing cancelling and resuming the task
2. Refactor the task list to not to be so unwieldy 

**Notes**
@Tobbe I didn't mean to undo any of your work on this and I probably don't have things as tidy as you did have them. I intend to further refactor things a little to get them back to being relatively neat. 

I also know we both didn't love the idea of just defining a huge task list. We need to think up an alternative approach if we want one.

Motivated by the fact that this task list has so many steps it was overflowing my fullscreen terminal window - the task list is currently making use of the alternative screen feature available to nearly all terminals. Again we probably should discuss the UX of this all if this ever ends up in any user facing code.

### CODE REVIEW

File: tasks/test-project/rebuild-test-project-fixture.js [hunk 1]
```diff
@@ -45,479 +71,660 @@ const args = yargs(hideBin(process.argv))
...
-          ...tuiTaskList[i],
-          parent: stepId,
-        })
-      }
+function indexIsLower(index, comparisonIndex) {
+  const levels1 = index.split('.')
+  const levels2 = comparisonIndex.split('.')
+  // Loop through the levels until one index is exhausted
+  for (let i = 0; i < Math.max(levels1.length, levels2.length); i++) {
+    const level1 = parseInt(levels1[i]) || 0 // Use 0 if level is undefined or NaN
```
Comment (Tobbe) [inline tasks/test-project/rebuild-test-project-fixture.js:140]: Just to be safe :) 
```suggestion
    const level1 = parseInt(levels1[i], 10) || 0 // Use 0 if level is undefined or NaN
```
And nice! I didn't know `NaN` was falsey!

File: tasks/test-project/rebuild-test-project-fixture.js [hunk 2]
```diff
@@ -45,479 +71,660 @@ const args = yargs(hideBin(process.argv))
...
+    const level2 = parseInt(levels2[i]) || 0 // Use 0 if level is undefined or NaN
```
Comment (Tobbe) [inline tasks/test-project/rebuild-test-project-fixture.js:141]:
```suggestion
    const level2 = parseInt(levels2[i], 10) || 0 // Use 0 if level is undefined or NaN
```

**Comment by Tobbe** [comment]
Great work Josh! And I'm excited about your vision on this 🙂 

I've scanned through the code, but it's a lot to take in 😅 So I haven't done a proper review of it all.

> Motivated by the fact that this task list has so many steps it was overflowing my fullscreen terminal window - the task list is currently making use of the alternative screen feature available to nearly all terminals. Again we probably should discuss the UX of this all if this ever ends up in any user facing code.

I have no idea what this "alternative screen feature" is. Would be great if you could show me on a video call.

**Comment by Josh-Walker-GM** [comment]
This is not a priority now, perhaps in the future we will update in this direction.

**Pull request closed without merging**